packages/RichTextEditor/src/RichTextEditor.tsx

```diff
@@ -0,0 +1,273 @@
+import LexicalErrorBoundary from '@lexical/react/LexicalErrorBoundary';
+// eslint-disable-next-line max-len
+import { CharacterLimitPlugin } from '@lexical/react/LexicalCharacterLimitPlugin';
+import { ListPlugin } from '@lexical/react/LexicalListPlugin';
+// eslint-disable-next-line max-len
+import { MarkdownShortcutPlugin } from '@lexical/react/LexicalMarkdownShortcutPlugin';
+import { ClearEditorPlugin } from '@lexical/react/LexicalClearEditorPlugin';
+import { OnChangePlugin } from '@lexical/react/LexicalOnChangePlugin';
+
+import { HeadingNode, QuoteNode } from '@lexical/rich-text';
+import { ListItemNode, ListNode } from '@lexical/list';
+import { AutoLinkNode, LinkNode } from '@lexical/link';
+import { OverflowNode } from '@lexical/overflow';
+import { CodeNode, CodeHighlightNode } from '@lexical/code';
+
+import { TRANSFORMERS } from '@lexical/markdown';
+
+import Lock from '@igloo-ui/icons/dist/Lock';
+import ToolbarPlugin from './plugins/ToolbarPlugin';
+import { OnFocusPlugin } from './plugins/OnFocusPlugin';
+import { DisablePlugin } from './plugins/DisablePlugin';
+import { MaxLengthPlugin } from './plugins/MaxLengthPlugin';
+import FloatingLinkEditorPlugin from './plugins/FloatingLinkEditorPlugin';
+import LinkPlugin from './plugins/LinkPlugin';
+import CodeHighlightPlugin from './plugins/CodeHighlightPlugin';
+
+import EditorTheme from './themes/RichTextEditor.theme';
+
+import './rich-text-editor.scss';
+
+export interface MessageOptions {
+  text?: string;
+  tooltip?: string;
+}
+
+export interface Messages {
+  bold?: MessageOptions;
+  clear?: MessageOptions;
+  italic?: MessageOptions;
+  link?: MessageOptions;
+  orderedList?: MessageOptions;
+  strikethrough?: MessageOptions;
+  underline?: MessageOptions;
+  unorderedList?: MessageOptions;
+  private?: MessageOptions;
+  linkEditorEdit?: MessageOptions;
+  linkEditorRemove?: MessageOptions;
+  linkEditorCancel?: MessageOptions;
+  linkEditorSave?: MessageOptions;
+}
+
+export interface RichTextEditorProps
+  extends Omit<React.ComponentProps<'div'>, 'onChange' | 'onFocus' | 'onBlur'> {
+  /** Whether or not the editor should be focused on load */
+  autoFocus?: boolean;
+  /** Add a class name to the rich text editor */
+  className?: string;
+  /** Add a data-test tag for automated tests */
+  dataTest?: string;
+  /** Whether or not the rich text editor is disabled */
+  disabled?: boolean;
+  /** Whether or not the right text editor has an error */
+  error?: boolean;
+  /** The initial editor state, which will preload the editor with content */
+  initialState?: string;
+  /** Whether or not the text can be cleared */
+  isClearable?: boolean;
+  /** Whether or not the rich text editor should have a private look */
+  isPrivate?: boolean;
+  /** The maximum amount or characters allowed
+   * (Automatically adds a character limit indicator) */
+  maxLength?: number;
+  /** Messages for the buttons */
```

Comment:
You should be a little more explicit on the description, at first I thought it was the text for the save button 🙃